core/src/test/java/org/elasticsearch/index/query/PercolatorQueryBuilderTests.java

```diff
@@ -75,6 +79,16 @@ private PercolatorQueryBuilder doCreateTestQueryBuilder(boolean indexedDocument)
         }
     }
 
+    /**
```

Comment:
Maybe the comment isn't clear hear. This PR changes the roundtrip test for xContent -> parsing -> builder so that all fields inside the query get randomly reordered. This also applies recursively to inner Objects. So when in the original builder you have ` "document" : { "foo" : "a", "bar" : "b" }` but you might get ` "document" : { "bar" : "b", "foo" : "a" }` in the parsed builder. Both documents are the same, but their bytes representation, which is stored in the builder, is different. This is why that fields inside "document" should not recursively be shuffled in the test.